src/ethereum_optimized/__init__.py

```diff
@@ -13,46 +13,77 @@
 
 They can be monkey patched in during start up by calling the `monkey_patch()`
-function.
+function. This must be done before those modules are imported anywhere.
 """
 
-from typing import Optional
+from importlib import import_module
+from typing import Any, Optional, cast
+
+from .fork import get_optimized_pow_patches
+from .state_db import get_optimized_state_patches
+
+
+def monkey_patch_optimized_state_db(
+    fork_name: str, state_path: Optional[str]
+) -> None:
+    """
+    Replace the state interface with one that supports high performance
+    updates and storing state in a database.
+
+    This function must be called before the state interface is imported
+    anywhere.
+    """
+    slow_state = cast(Any, import_module("ethereum." + fork_name + ".state"))
+
+    optimized_state_db_patches = get_optimized_state_patches(fork_name)
+
+    for (name, value) in optimized_state_db_patches.items():
+        setattr(slow_state, name, value)
+
+    if state_path is not None:
+        slow_state.State.default_path = state_path
+
+
+def monkey_patch_optimized_spec(fork_name: str) -> None:
+    """
+    Replace the ethash implementation with one that supports higher
+    performance.
+
+    This function must be called before the spec interface is imported
+    anywhere.
+    """
+    slow_spec = import_module("ethereum." + fork_name + ".fork")
+
+    optimized_pow_patches = get_optimized_pow_patches(fork_name)
+
+    for (name, value) in optimized_pow_patches.items():
+        setattr(slow_spec, name, value)
 
 
 def monkey_patch(state_path: Optional[str]) -> None:
     """
     Apply all monkey patches to the specification.
     """
-    from . import (
-        arrow_glacier,
-        berlin,
-        byzantium,
-        constantinople,
-        dao_fork,
-        frontier,
-        gray_glacier,
-        homestead,
-        istanbul,
-        london,
-        muir_glacier,
-        paris,
-        shanghai,
-        spurious_dragon,
-        tangerine_whistle,
-    )
-
-    frontier.monkey_patch(state_path)
-    homestead.monkey_patch(state_path)
-    dao_fork.monkey_patch(state_path)
-    tangerine_whistle.monkey_patch(state_path)
-    spurious_dragon.monkey_patch(state_path)
-    byzantium.monkey_patch(state_path)
-    constantinople.monkey_patch(state_path)
-    istanbul.monkey_patch(state_path)
-    muir_glacier.monkey_patch(state_path)
-    berlin.monkey_patch(state_path)
-    london.monkey_patch(state_path)
-    arrow_glacier.monkey_patch(state_path)
-    gray_glacier.monkey_patch(state_path)
-    paris.monkey_patch(state_path)
-    shanghai.monkey_patch(state_path)
+    for fork_name in (
+        "frontier",
+        "homestead",
+        "dao_fork",
+        "tangerine_whistle",
+        "spurious_dragon",
+        "byzantium",
+        "constantinople",
+        "istanbul",
+        "muir_glacier",
+        "berlin",
+        "london",
+        "arrow_glacier",
+        "gray_glacier",
+    ):
```

Comment:
Is there some way we can put this information into the fork itself and retrieve it using functions from [`ethereum_spec_tools/forks.py`](https://github.com/ethereum/execution-specs/blob/master/src/ethereum_spec_tools/forks.py)?

I _really_ want to keep metadata about forks (like these pre-/post-merge lists) in as few places as possible.

Reply:
Having thought about this it is possible and I have done it.